galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/TestClassesRas.java

```diff
@@ -23,45 +24,69 @@
 
 @Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
-        "osgi.http.whiteboard.servlet.pattern=/ras/testclasses" }, name = "TestClasses RAS")
+"osgi.http.whiteboard.servlet.pattern=/ras/testclasses" }, name = "TestClasses RAS")
 
 public class TestClassesRas extends HttpServlet {
 
-    /**
-     *
-     */
-    private static final long serialVersionUID = 1L;
-
-    /* a dummy list of testclasses and bundles */
-    private final List<RasTestClass> TESTCLASSDUMMY = Arrays.asList(new RasTestClass("FirstTest", "Abundle"),new RasTestClass("SecondTest", "BigBundle"),new RasTestClass("ThirdTest", "MiniBundle"),new RasTestClass("ZeroTest", "NanoBundle"));
-    
-
-    @Reference
-    public IFramework framework; // NOSONAR
-
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        Map<String, String[]> query = req.getParameterMap();
-
-        /* looking for sort options in query and sorting accordingly */
-        if(ExtractQuerySort.isAscending(query, "testclass")) {
-    		TESTCLASSDUMMY.sort(Comparator.comparing(RasTestClass::getTestClass));
-    	}else if(!ExtractQuerySort.isAscending(query, "testclass")) {
-    		TESTCLASSDUMMY.sort(Comparator.comparing(RasTestClass::getTestClass).reversed());
-        }
-        /* converting data to json */
-            JsonElement json = new Gson().toJsonTree(TESTCLASSDUMMY);
-            JsonObject testclasses = new JsonObject();
-            testclasses.add("testclasses", json);
-            
-        /* setting response status and type */
-            resp.setStatus(200);
-            resp.setContentType("application/json");
-            PrintWriter out = resp.getWriter();
-            out.print(testclasses);
-        
-    }
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1L;
+
+	private List<IResultArchiveStoreDirectoryService> archiveStore;
+
+	private IResultArchiveStoreDirectoryService directoryService;
+
+	private static List<RasTestClass> classArray = new ArrayList<>();
+	/* a dummy list of testclasses and bundles */
+	private final List<RasTestClass> TESTCLASSDUMMY = Arrays.asList(new RasTestClass("FirstTest", "Abundle"),
+			new RasTestClass("SecondTest", "BigBundle"), new RasTestClass("ThirdTest", "MiniBundle"),
+			new RasTestClass("ZeroTest", "NanoBundle"));
+
+	@Reference
+	public IFramework framework; // NOSONAR
+
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		Map<String, String[]> query = req.getParameterMap();
+		archiveStore = framework.getResultArchiveStore().getDirectoryServices();
+		for(IResultArchiveStoreDirectoryService directory: archiveStore){
+			directoryService=directory;
+		}
+		try{
+			classArray = directoryService.getTests();
+		} catch (ResultArchiveStoreException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
if this exception is thrown we should respond to the get request with a status != 200